File: example-java/src/main/java/io/github/cmh1448/examplejava/dto/AuthDto.java
package io.github.cmh1448.examplejava.dto;

import io.github.cmh1448.examplejava.model.User;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.security.crypto.password.PasswordEncoder;

public class AuthDto {

    @AllArgsConstructor
    @NoArgsConstructor
    @Data
    @Builder
    public static class Request {
        private String id;
        private String password;

        public User toUser(PasswordEncoder encoder) {
            return User.builder()
                    .id(id)
                    .password(encoder.encode(password))
                    .build();
        }
    }
}

File: example/src/main/kotlin/io/github/cmh1448/autojwt/example/dto/AuthDto.kt
package io.github.cmh1448.autojwt.example.dto

import io.github.cmh1448.autojwt.example.model.User
import org.springframework.security.crypto.password.PasswordEncoder

class AuthDto {
    data class Request(
        val id: String,
        val password: String,
    ) {
        fun toUser(
            passwordEncoder: PasswordEncoder
        ): User {
            return User(
                id, passwordEncoder.encode(password)
            )
        }
    }
}